src/VirtualFile/VirtualPath.cs

```diff
@@ -3,12 +3,15 @@
 using System.Text;
 using System.Web.Hosting;
 using System.Web.Util;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.FileProviders;
 
 namespace System.Web;
 
-internal sealed class VirtualPath
+public sealed class VirtualPath
```

Comment:
keep internal

Making anything public that was not already public in webforms will warrant its own discussion :)